### ISSUE TITLE
PHP 8.1: New #[ReturnTypeWillChange] attribute

### ISSUE THREAD

**Issue by scottgruber** [schmittjoh/php-option#61]
Hi,

I came across an error after updating to PHP 8.1 while loading a via CraftCMS site locally. The same error also happened using PHP8.0. 

[PHPWatch posted a new PHP8.1 attribute `New #[ReturnTypeWillChange] attribute`](https://php.watch/versions/8.1/ReturnTypeWillChange) which may solve the issue. 

Here is the error.

```
Deprecated: Return type of PhpOption\Some::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /vendor/phpoption/phpoption/src/PhpOption/Some.php on line 152

Deprecated: Return type of PhpOption\None::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /vendor/phpoption/phpoption/src/PhpOption/None.php on line 118
```

**Some.php Before**

```
    public function getIterator()
    {
        return new ArrayIterator([$this->value]);
    }
```

**Some.php After**

```
 #[\ReturnTypeWillChange]
    public function getIterator()
    {
        return new ArrayIterator([$this->value]);
    }
```

**None.php Before**

```
public function getIterator()
    {
        return new EmptyIterator();
    }

```

**None.php After**

```
 #[\ReturnTypeWillChange]
 public function getIterator()
    {
        return new EmptyIterator();
    }

```

As a quick test, I took a look in AbstractVariables.php and added `#[\ReturnTypeWillChange]` before the public function declaration and the warning was suppressed. I read your contributing guidelines, and could try to make the changes and submit a PR, but I'm a junior PHP developer and might not code it correctly and there are many public function declarations that I may miss.

**Comment by jdreesen** [comment]
Are you sure you're using the latest version? It should have been fixed in `v1.8.0`. See: #59

**Comment by scottgruber** [comment]
Fixed. I removed the composer.lock file and ran `composer install` in my craft installation and that updated php-option to v1.8.0. I'll let CraftCMS dev community know in their discord chat channel.
